src/string/_internal/getWords.ts

```diff
@@ -16,6 +16,81 @@
  */
 const CASE_SPLIT_PATTERN = /[A-Z]?[a-z]+|[0-9]+|[A-Z]+(?![a-z])/g;
 
-export function getWords(str: string): string[] {
-  return Array.from(str.match(CASE_SPLIT_PATTERN) ?? []);
+export function getWords<T extends string>(str: T): CaseSplit<T> {
+  return Array.from(str.match(CASE_SPLIT_PATTERN) ?? []) as CaseSplit<T>;
 }
+
+type UppercaseLetter =
+  | 'A'
+  | 'B'
+  | 'C'
+  | 'D'
+  | 'E'
+  | 'F'
+  | 'G'
+  | 'H'
+  | 'I'
+  | 'J'
+  | 'K'
+  | 'L'
+  | 'M'
+  | 'N'
+  | 'O'
+  | 'P'
+  | 'Q'
+  | 'R'
+  | 'S'
+  | 'T'
+  | 'U'
+  | 'V'
+  | 'W'
+  | 'X'
+  | 'Y'
+  | 'Z';
+type LowercaseLetter = UppercaseLetter extends `${infer U}` ? Lowercase<U> : never;
```

Comment:
why do you need the `extends` here?

```suggestion
type LowercaseLetter = Lowercase<UppercaseLetter>;
```